lib/internal/test_runner/test.js

```diff
@@ -344,6 +346,14 @@ class Test extends AsyncResource {
     }
   }
 
+  removeSubtest(subtest) {
```

Comment:
Is this something that can live in `FileTest` in runner.js?

Reply:
it cannot be a method on `FileTest` since we call `root.removeSubtest(subtest)` and `root` is not a `FileTest`,
the reason why I set this logic inside a method of `Test` is that the logic behind `waitingOn` is managed solely inside `Test` and is not really the concern of `FileTest`

Reply:
> the logic behind waitingOn is managed solely inside Test and is not really the concern of FileTest

Since `FileTest` inherits from `Test` it is free to do things with its `waitingOn` too. I was hoping to keep `Test` as minimal as possible because as things keep getting added there, the test runner slows down. More importantly though, I think this new method is only correct in the context of `FileTest` and watch mode due to what it is doing with `waitingOn`. By putting the logic in `Test`, that is inviting someone in the future to try using it and I don't believe it will work as expected.

Reply:
I'v pushed a change, I think the code is less readable this way, but I do understand your concern regarding incorrectly using that method